sycl/plugins/unified_runtime/ur/adapters/cuda/device.cpp

```diff
@@ -0,0 +1,1183 @@
+    return ReturnValue(result);
+  }
+  case UR_DEVICE_INFO_HOST_UNIFIED_MEMORY: {
+    int is_integrated = 0;
+    sycl::detail::ur::assertion(
+        cuDeviceGetAttribute(&is_integrated, CU_DEVICE_ATTRIBUTE_INTEGRATED,
+                             device->get()) == CUDA_SUCCESS);
+
+    sycl::detail::ur::assertion((is_integrated == 0) | (is_integrated == 1));
+    auto result = static_cast<bool>(is_integrated);
+    return ReturnValue(result);
+  }
+  case UR_DEVICE_INFO_PROFILING_TIMER_RESOLUTION: {
+    // Hard coded to value returned by clinfo for OpenCL 1.2 CUDA | GeForce GTX
+    // 1060 3GB
+    return ReturnValue(1000lu);
+  }
+  case UR_DEVICE_INFO_ENDIAN_LITTLE: {
+    return ReturnValue(true);
+  }
+  case UR_DEVICE_INFO_AVAILABLE: {
+    return ReturnValue(true);
+  }
+  case UR_DEVICE_INFO_BUILD_ON_SUBDEVICE: {
+    return ReturnValue(true);
+  }
+  case UR_DEVICE_INFO_COMPILER_AVAILABLE: {
+    return ReturnValue(true);
+  }
+  case UR_DEVICE_INFO_LINKER_AVAILABLE: {
+    return ReturnValue(true);
+  }
+  case UR_DEVICE_INFO_EXECUTION_CAPABILITIES: {
+    auto capability = ur_device_exec_capability_flags_t{
+        UR_DEVICE_EXEC_CAPABILITY_FLAG_KERNEL};
+    return ReturnValue(capability);
+  }
+  case UR_DEVICE_INFO_QUEUE_PROPERTIES:
+    return ReturnValue(
+        ur_queue_flag_t(UR_QUEUE_FLAG_OUT_OF_ORDER_EXEC_MODE_ENABLE |
+                        UR_QUEUE_FLAG_PROFILING_ENABLE));
+  case UR_DEVICE_INFO_QUEUE_ON_DEVICE_PROPERTIES: {
+    // The mandated minimum capability:
+    uint64_t capability = UR_QUEUE_FLAG_PROFILING_ENABLE |
+                          UR_QUEUE_FLAG_OUT_OF_ORDER_EXEC_MODE_ENABLE;
+    return ReturnValue(capability);
+  }
+  case UR_DEVICE_INFO_QUEUE_ON_HOST_PROPERTIES: {
+    // The mandated minimum capability:
+    uint64_t capability = UR_QUEUE_FLAG_PROFILING_ENABLE;
+    return ReturnValue(capability);
+  }
+  case UR_DEVICE_INFO_BUILT_IN_KERNELS: {
+    // An empty string is returned if no built-in kernels are supported by the
+    // device.
+    return ReturnValue("");
+  }
+  case UR_DEVICE_INFO_PLATFORM: {
+    return ReturnValue(device->get_platform());
+  }
+  case UR_DEVICE_INFO_NAME: {
+    static constexpr size_t MAX_DEVICE_NAME_LENGTH = 256u;
+    char name[MAX_DEVICE_NAME_LENGTH];
+    sycl::detail::ur::assertion(cuDeviceGetName(name, MAX_DEVICE_NAME_LENGTH,
+                                                device->get()) == CUDA_SUCCESS);
+    return ReturnValue(name, strlen(name) + 1);
+  }
+  case UR_DEVICE_INFO_VENDOR: {
+    return ReturnValue("NVIDIA Corporation");
+  }
+  case UR_DEVICE_INFO_DRIVER_VERSION: {
+    auto version = getCudaVersionString();
+    return ReturnValue(version.c_str());
+  }
+  case UR_DEVICE_INFO_PROFILE: {
+    return ReturnValue("CUDA");
+  }
+  case UR_DEVICE_INFO_REFERENCE_COUNT: {
+    return ReturnValue(device->get_reference_count());
+  }
+  case UR_DEVICE_INFO_VERSION: {
+    return ReturnValue("PI 0.0");
```

Comment:
```suggestion
    return ReturnValue("UR 0.0");
```
?

Reply:
Good question. This is a holder over from OpenCL and underspecified in UR.

I think I'm inclined to suggest this should not change until we completely remove PI.

Reply:
This now returns the actual CUDA version